corehq/warehouse/migrations/0004_app_status_fact_form_and_sync_staging_and_deleted_field.py

```diff
@@ -0,0 +1,103 @@
+        ),
+        migrations.CreateModel(
+            name='FormStagingTable',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('form_id', models.CharField(max_length=255, unique=True)),
+                ('domain', models.CharField(default=None, max_length=255)),
+                ('app_id', models.CharField(max_length=255, null=True)),
+                ('xmlns', models.CharField(default=None, max_length=255)),
+                ('user_id', models.CharField(max_length=255, null=True)),
+                ('received_on', models.DateTimeField(db_index=True)),
+                ('build_id', models.CharField(max_length=255, null=True)),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='SyncLogStagingTable',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('sync_log_id', models.CharField(max_length=255)),
+                ('sync_date', models.DateTimeField(null=True)),
+                ('domain', models.CharField(max_length=255, null=True)),
+                ('user_id', models.CharField(max_length=255, null=True)),
+                ('build_id', models.CharField(max_length=255, null=True)),
+                ('duration', models.IntegerField(null=True)),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.RenameField(
+            model_name='userdim',
+            old_name='user_type',
+            new_name='doc_type',
+        ),
+        migrations.AddField(
+            model_name='domaindim',
+            name='deleted',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='groupdim',
+            name='deleted',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='locationdim',
+            name='deleted',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='userdim',
+            name='deleted',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='usergroupdim',
+            name='deleted',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='userlocationdim',
+            name='deleted',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='applicationstatusfact',
+            name='user_dim',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='warehouse.UserDim'),
```

Comment:
@snopoke the foreignkey is down here strangely. i'll autogenerate the migration from scratch again once it's good to go